akka-actor-typed/src/main/scala/akka/actor/typed/internal/TimerSchedulerImpl.scala

```diff
@@ -75,9 +67,9 @@ import scala.concurrent.duration.FiniteDuration
 
     val timerMsg =
       if (msg.isInstanceOf[NotInfluenceReceiveTimeout])
-        NotInfluenceReceiveTimeoutTimerMsg(key, nextGen, this)
+        new TimerMsg(key, nextGen, this) with NotInfluenceReceiveTimeout
```

Comment:
`case class` should preferably be `final` according to normal Scala conventions, so perhaps make it ordinary `class` and implement `toString`